docs/ai/semantic-kernel-dotnet-overview.md

```diff
@@ -0,0 +1,217 @@
+- Improves reliability by reducing the unpredictable behavior of prompts and responses from AI models. You can fine-tune prompts and plan tasks to create a controlled and predictable user experience.
+
+Semantic Kernel is built around several core concepts:
+
+- **Connections**: Interface with external AI services and data sources.
+- **Plugins**: Encapsulate functions that applications can use.
+- **Planner**: Orchestrates execution plans and strategies based on user behavior.
+- **Memory**: Abstracts and simplifies context management for AI apps.
+
+These building blocks are explored in more detail in the following sections.
+
+### Connections
+
+The Semantic Kernel SDK includes a set of connectors that enable developers to integrate LLMs and other services into their existing applications. These connectors serve as the bridge between the application code and the AI models or services. Semantic Kernel handles many common connection concerns and challenges for you so you can focus on building your own workflows and features.
+
+The following code snippet creates a `Kernel` and adds a connection to an Azure OpenAI model:
+
+```csharp
+using Microsoft.SemanticKernel;
+
+// Create kernel
+var builder = Kernel.CreateBuilder();
+
+// Add a chat completion service:
+builder.Services.AddAzureOpenAIChatCompletion(
+    "your-resource-name",
+    "your-endpoint",
+    "your-resource-key",
+    "deployment-model");
+var kernel = builder.Build();
+```
+
+### Plugins
+
+Semantic Kernel [plugins](/semantic-kernel/agents/plugins) encapsulate standard language functions for applications and AI models to consume. You can create your own plugins or rely on plugins provided by the SDK. These plugins streamline tasks where AI models are advantageous and efficiently combine them with more traditional C# methods. Plugin functions are generally categorized into two types, which are semantic functions and native functions.
+
+#### Semantic functions
+
+Semantic functions are essentially AI prompts defined in your code that Semantic Kernel can customize and call as needed. These prompts can be templatized to use variables, custom prompt and completion  formatting, and more.
+
+The following code snippet defines and registers a semantic function:
+
+```csharp
+var userInput = Console.ReadLine();
+
+//Define semantic function inline
+string skPrompt = @"Summarize the provided unstructured text in a sentence that is easy to understand.
+                    Text to summarize: {{$userInput}}";
+
+// Register the function
+kernel.CreateSemanticFunction(
+    promptTemplate: skPrompt, 
+    functionName: "SummarizeText",
+    pluginName: "SemanticFunctions"
+);
+```
+
+#### Native functions
+
+Native functions are C# methods that Semantic Kernel can call directly to manipulate or retrieve data, and perform other operations that are better suited for traditional code instructions instead of LLM prompts.
```

Comment:
```suggestion
Native functions are C# methods that Semantic Kernel can call directly to manipulate or retrieve data. They perform operations that are better suited for traditional code instructions instead of LLM prompts.
```